tests/Listener/ShareCreatedEventListenerTest.php

```diff
@@ -96,12 +181,159 @@ public function testShareTypeEmailShallHandled(): void
             ->method('getNode')
             ->willReturn($this->mockNode);
 
-        $this->mockLogger->expects($this->exactly(1))
-            ->method('debug')
+        $this->mockShare
+            ->method('getToken')
+            ->willReturn(ShareCreatedEventListenerTest::MOCK_SHARE_TOKEN);
+
+        $this->mockShare
+            ->method('getSharedWith')
+            ->willReturn($mockRecipient);
+
+        $this->mockShare
+            ->method('getNote')
+            ->willReturn($mockNote);
+
+
+        if ($mockExpiration !== null) {
+            $this->mockShare
+                ->method('getExpirationDate')
+                ->willReturn($this->mockTimestamp);
+
+            $this->mockTimestamp
+                ->method('getTimestamp')
+                ->willReturn($mockExpiration);
+        } else {
+            $this->mockShare
+                ->method('getExpirationDate')
+                ->willReturn(null);
+        }
+    }
+
+    public function testShareTypeEmailNoExpirationDate(): void
+    {
+        $mockLanguageCode = 'lang_LOCALE';
+        $mockNodeName = "mock-name";
+        $mockUrl = "mock-url";
+        $mockExpiration = null;
+        $mockRecipient = "mock-recipient";
+
+        $this->configureMocks(
+            true,
```

Comment:
suggestion: to make this easier to read userFound could also be a var here

Reply:
Solved by adding an enum